tests/Casts/BooleanTest.php

```diff
@@ -0,0 +1,58 @@
+<?php
+
+declare(strict_types=1);
+
+namespace MongoDB\Laravel\Tests\Casts;
+
+use MongoDB\Laravel\Tests\Models\Casting;
+use MongoDB\Laravel\Tests\TestCase;
+
+class BooleanTest extends TestCase
+{
+    protected function setUp(): void
+    {
+        parent::setUp();
+
+        Casting::truncate();
+    }
+
+    public function testBool(): void
+    {
+        $model = Casting::query()->create(['booleanValue' => true]);
+
+        self::assertIsBool($model->booleanValue);
+        self::assertEquals(true, $model->booleanValue);
```

Comment:
It is preferable to use the strict `assertSame` when possible.

Reply:
Done in fdaa9dc.